sklearn/linear_model/tests/test_ridge.py

```diff
@@ -311,6 +311,96 @@ def test_ridge_individual_penalties():
     assert_raises(ValueError, ridge.fit, X, y)
 
 
+@pytest.mark.parametrize('gcv_mode', ['svd', 'eigen'])
+@pytest.mark.parametrize('X_constructor', [np.asarray, sp.csr_matrix])
+@pytest.mark.parametrize('X_shape', [(11, 8), (11, 20)])
+@pytest.mark.parametrize('fit_intercept', [True, False])
+@pytest.mark.parametrize('normalize', [True, False])
+@pytest.mark.parametrize('noise', [1., 30.])
+def test_ridge_gcv_vs_k_fold(gcv_mode, X_constructor, X_shape, fit_intercept,
+                             normalize, noise):
+    n_samples, n_features = X_shape
+    X, y = make_regression(
+        n_samples=n_samples, n_features=n_features, n_targets=3,
+        random_state=0, shuffle=False, noise=noise, n_informative=5
+    )
+    X += 30 * np.random.RandomState(0).randn(X.shape[1])
+
+    alphas = [1e-3, .1, 1., 10., 1e3]
+    loo_ridge = RidgeCV(cv=n_samples, fit_intercept=fit_intercept,
+                        alphas=alphas, scoring='neg_mean_squared_error',
+                        normalize=normalize)
+    gcv_ridge = RidgeCV(gcv_mode=gcv_mode, fit_intercept=fit_intercept,
+                        alphas=alphas, normalize=normalize)
+
+    loo_ridge.fit(X, y)
+
+    X_gcv = X_constructor(X)
+    gcv_ridge.fit(X_gcv, y)
+
+    assert gcv_ridge.alpha_ == pytest.approx(loo_ridge.alpha_)
+    assert_allclose(gcv_ridge.coef_, loo_ridge.coef_, rtol=1e-4)
+    assert_allclose(gcv_ridge.intercept_, loo_ridge.intercept_, rtol=1e-4)
```

Comment:
Very good test!

Reply:
I believe that there is [no example](https://scikit-learn.org/dev/modules/generated/sklearn.linear_model.RidgeCV.html#examples-using-sklearn-linear-model-ridgecv) that highlights the use and speed of GCV. I think it would be great to write one to show the equivalence between such an instance of `loo_ridge` and `gcv_ridge` on a small regression dataset such as Boston for instance.

One could plot the information collected with `store_cv_values=True`.

Finally one could also compare to the result obtained with 10-fold cross-validation (on neg MSE too).